### PR TITLE
Refactor: restructure component layers for kebab button on groupcard

### LINKED ISSUE #385: Restructure component layers for Kebab button on GroupCard
## Description

this issue is child of #345

### PR DESCRIPTION
**Pull request by ShoeheyOt**
## Overview

This PR is for refactoring, changing structure 

## Changes

The below 2 diagrams are comparison of how I changed component structure.
Before
```mermaid
graph TD;
GroupCard --> GroupCardContent;
GroupCardContent --> RenameGroupForm & Card;
Card --> GroupCardDropdownMenuTriggerButton;
GroupCardDropdownMenuTriggerButton --> DropdownMenu;
DropdownMenu --> GroupCardDropdownMenuContent;
GroupCardDropdownMenuContent --> DropdownMenuContent;
DropdownMenuContent --> DeleteGroupDialogTriggerButton;
DeleteGroupDialogTriggerButton --> DialogRoot;
DialogRoot --> DeleteGroupDialogContent;
```

After
```mermaid
graph TD;
GroupCard --> GroupCardContent;
GroupCardContent --> RenameGroupForm & Card;
Card --> GroupCardDropdownMenu;
GroupCardDropdownMenu --> DropdownMenu & DialogRoot;
DropdownMenu --> GroupCardDropdownMenuContent;
DialogRoot --> DeleteGroupDialogContent
```

## Screen Captures

[screen-capture.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/a02413a8-45bc-4034-adeb-7f4c8510ac03)

## Assignee Checklist:

- [X] The naming convention of the PR title is correct (See the comment at the top of this template)
- [X] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [X] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [X] The correct assignees and reviewers have been designated for this PR
- [X] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [X] All the related issues are associated with this PR
- [X] All criteria in the associated issues are met (please tick the checkboxes)
- [X] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/index.tsx [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
+   * a group name used to display on each card
+   */
+  groupName: IGroup['name'];
+  /**
+   * a number of container which belongs to a group
+   */
   containerCount: number;
+  /**
+   * a number of user who belongs to a group
+   */
```
Comment (nickyito) [inline src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/index.tsx:28]: - The ID (identifier) of ...
- The number of ...

File: src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenu/GroupCardDropdownMenuContent.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  handleDeleteClick: () => void;
+}
+
+export const GroupCardDropdownMenuContent = ({
+  handleRenameClick,
+  handleDeleteClick,
+}: IGroupCardDropdownMenuContentProps) => {
+  return (
+    <DropdownMenuContent>
+      <DropdownMenuItem>
```
Comment (nickyito) [inline src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenu/GroupCardDropdownMenuContent.tsx]: `asChild` should be added so we can remove the unnecessary div and manipulate it only with the keyboard.

Do you mind fixing the ones on the profile page as well?

File: src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenu/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+export const GroupCardDropdownMenu = ({
+  groupId,
+  handleRenameClick,
+  isRenameFormOpen,
+}: IGroupCardMenuButtonProps) => {
+  const [isDropdownMenuOpen, setIsDropdownMenuOpen] = useState(false);
+  const [isDeleteGroupDialogOpen, setIsDeleteGroupDialogOpen] = useState(false);
+
+  const kebabRef = useAutoFocus<HTMLButtonElement>(!isDeleteGroupDialogOpen && !isRenameFormOpen);
```
Comment (nickyito) [inline src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenu/index.tsx:35]: Thank you for adding this!

**Review by nickyito — CHANGES_REQUESTED** [review]
@ShoeheyOt 
Thank you for your work and for clarifying the structure with Mermaid. And, I am sorry for my late review.

I suppose some tiny adjustments might be needed. Please take a look at the comments.

File: src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenu/GroupCardDropdownMenuContent.tsx (hunk 2 above)
Comment (ShoeheyOt) [reply src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenu/GroupCardDropdownMenuContent.tsx]: @nick-y-ito Thank you for your review! as we dicussed in person, I leave fixing ones on the profile page and make an another issue.

**Review by nickyito — APPROVED** [review]
@ShoeheyOt 
Thank you for updating! LGTM :)

**Pull request merged**